ui/admin/app/components/form/field/list-wrapper/key-value/index.js

```diff
@@ -29,6 +29,38 @@ export default class MappingListComponent extends Component {
   get options() {
     return this.args?.options || this.args?.model?.[this.args.name];
   }
+
+  /**
+   * Prevents users from selecting duplicate keys from the select list if there's the arg is set to true
+   * @type {object}
+   */
+
+  get selectOptions() {
+    const previouslySelectedKeys = this.options || [];
+    if (this.args.removeDuplicates && previouslySelectedKeys.length) {
+      return this.args.selectOptions.filter((key) =>
+        previouslySelectedKeys.every((obj) => obj.key !== key),
+      );
+    } else {
+      return this.args.selectOptions;
+    }
+  }
+
+  /**
+   * Determines if we need to show an empty row to the users to enter more key/value pairs based on removeDuplicates arg,
+   * by default it is true
+   * @type {object}
+   */
+  get showNewRow() {
+    if (this.args.removeDuplicates) {
+      return (
+        (this.options || []).length !== (this.args.selectOptions || []).length
```

Comment:
```suggestion
        (this.options ?? []).length !== (this.args.selectOptions ?? []).length
```

Does `this.options?.length` !== `this.args.selectOptions?.length` not work?

Reply:
no that totally works, I'll simplify this